src/rust/src/wallet_scanner.rs

```diff
@@ -768,10 +657,8 @@ fn init_batch_scanner(
         let ivks: Vec<(_, _)> = sapling_ivks
             .iter()
             .map(|raw_ivk| {
-                let ivk: Option<_> = jubjub::Fr::from_bytes(raw_ivk)
-                    .map(|scalar_ivk| PreparedIncomingViewingKey::new(&SaplingIvk(scalar_ivk)))
-                    .into();
-                ivk.map(|prepared_ivk| (*raw_ivk, prepared_ivk))
+                parse_and_prepare_sapling_ivk(raw_ivk)
+                    .map(|prepared_ivk| (*raw_ivk, prepared_ivk))
                     .ok_or("Invalid Sapling ivk passed to wallet::init_batch_scanner()")
```

Comment:
```suggestion
                    .ok_or("Invalid Sapling ivk passed to wallet_scanner::init_batch_scanner()")
```

Reply:
DItto: this error gets thrown on the C++ side, so is written from its perspective.